File: hilllistcore/src/main/java/uk/colessoft/android/hilllist/model/GeographImageSearchResponse.java
package uk.colessoft.android.hilllist.model;


import java.util.ArrayList;
import java.util.List;

public class GeographImageSearchResponse {

    public List<GeographImageDetail> getItems() {
        return items;
    }

    public void setItems(List<GeographImageDetail> items) {
        this.items = items;
    }

    private List<GeographImageDetail> items;

    public List<String> getThumbnailUrls() {

        List<String> thumbnailUrls = new ArrayList<>();

        for(GeographImageDetail detail:items){
            thumbnailUrls.add(detail.getThumb());
        }

        return thumbnailUrls;
    }
}

File: app/src/main/java/uk/colessoft/android/hilllist/domain/GeographImageSearchResponse.kt
package uk.colessoft.android.hilllist.domain


import java.util.ArrayList

class GeographImageSearchResponse {

    var items: List<GeographImageDetail>? = null

    val thumbnailUrls: List<String>
        get() {

            val thumbnailUrls = ArrayList<String>()

            for (detail in items!!) {
                thumbnailUrls.add(detail.thumb.orEmpty())
            }

            return thumbnailUrls
        }
}
